### PR TITLE
String.xml -- Try .NET -- replacing out-of-sync PR 3458

### PR DESCRIPTION
**Pull request by WilliamAntonRohm**
new samples PR is https://github.com/dotnet/samples/pull/1916

replaces https://github.com/dotnet/dotnet-api-docs/pull/3458

started with an up-to-date copy of String.xml & re-interactivated sample file references

### CODE REVIEW

**Comment by WilliamAntonRohm** [comment]
@mairaw -- Maira, please review this (hopefully last of this series) version of String.xml, which includes all the new interactive sample refs.

File: xml/System/String.xml [hunk 1]
```diff
@@ -14710,12 +14710,12 @@ The `TrimStart(System.Char[])` method removes from the current string all leadin
 
 The following example demonstrates the basic functionality of the <xref:System.String.TrimStart%2A> method:
 
-[!code-csharp[System.String.TrimStart#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs#1)]
+[!code-csharp-interactive[System.String.TrimStart#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs#1)]
```
Comment (mairaw) [inline xml/System/String.xml]: there's a bug with try.net with this one

```suggestion
[!code-csharp[System.String.TrimStart#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs#1)]
```

File: xml/System/String.xml [hunk 2]
```diff
@@ -14710,12 +14710,12 @@ The `TrimStart(System.Char[])` method removes from the current string all leadin
...
 [!code-vb[System.String.TrimStart#1](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.TrimStart/vb/sample.vb#1)]
 
 The following example uses the <xref:System.String.TrimStart%2A> method to trim white space and comment characters from lines of source code. The `StripComments` method wraps a call to <xref:System.String.TrimStart%2A> and passes it a character array that contains a space and the comment character, which is an apostrophe ( ' ) in Visual Basic and a slash ( / ) in C#. The <xref:System.String.TrimStart%2A> method is also called to remove leading white space when evaluating whether a string is a comment.
 
-[!code-csharp[System.String.TrimStart#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs#2)]
+[!code-csharp-interactive[System.String.TrimStart#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs#2)]
```
Comment (mairaw) [inline xml/System/String.xml]: this one doesn't look runnable 

```suggestion
[!code-csharp[System.String.TrimStart#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimStart/cs/sample.cs#2)]
```

File: xml/System/String.xml [hunk 3]
```diff
@@ -8022,7 +8022,7 @@ The `comparisonType` parameter is a <xref:System.StringComparison> enumeration m
  <xref:System.String.IsNullOrEmpty%2A> is a convenience method that enables you to simultaneously test whether a <xref:System.String> is `null` or its value is <xref:System.String.Empty?displayProperty=nameWithType>. It is equivalent to the following code:
 
  [!code-cpp[System.String.IsNullOrEmpty#1](~/samples/snippets/cpp/VS_Snippets_CLR_System/system.string.isnullorempty/cpp/isnullorempty1.cpp#1)]
- [!code-csharp[System.String.IsNullOrEmpty#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.string.isnullorempty/cs/isnullorempty1.cs#1)]
+ [!code-csharp-interactive[System.String.IsNullOrEmpty#1](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.string.isnullorempty/cs/isnullorempty1.cs#1)]
```
[the remark this answers is not in the document]
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml [hunk 4]
```diff
@@ -2271,7 +2271,7 @@ Examples of instantiating strings:
  The following example performs two comparisons of two substrings that only differ in case. The first comparison ignores case and the second comparison considers case.
 
  [!code-cpp[string.compare4#1](~/samples/snippets/cpp/VS_Snippets_CLR/string.compare4/CPP/comp4.cpp#1)]
- [!code-csharp[string.compare4#1](~/samples/snippets/csharp/VS_Snippets_CLR/string.compare4/CS/comp4.cs#1)]
+ [!code-csharp-interactive[string.compare4#1](~/samples/snippets/csharp/VS_Snippets_CLR/string.compare4/CS/comp4.cs#1)]
```
[the remark this answers is not in the document]
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml (hunk 2 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml [hunk 5]
```diff
@@ -6645,7 +6645,7 @@ The `GetPinnableReference` method returns a character that can be used for pinni
...
 
- [!code-csharp[System.String.IndexOf#21](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.IndexOf/CS/ignorable21.cs#21)]
+ [!code-csharp-interactive[System.String.IndexOf#21](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.IndexOf/CS/ignorable21.cs#21)]
```
[the remark this answers is not in the document]
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml [hunk 6]
```diff
@@ -7112,7 +7112,7 @@ The `comparisonType` parameter is a <xref:System.StringComparison> enumeration m
  The following example finds the index of all occurrences of the string "he" within a substring of another string. Note that the number of characters to be searched must be recalculated for each search iteration.
 
  [!code-cpp[string.indexof8#1](~/samples/snippets/cpp/VS_Snippets_CLR/string.indexof8/CPP/ixof8.cpp#1)]
- [!code-csharp[string.indexof8#1](~/samples/snippets/csharp/VS_Snippets_CLR/string.indexof8/CS/ixof8.cs#1)]
+ [!code-csharp-interactive[string.indexof8#1](~/samples/snippets/csharp/VS_Snippets_CLR/string.indexof8/CS/ixof8.cs#1)]
  [!code-vb[string.indexof8#1](~/samples/snippets/visualbasic/VS_Snippets_CLR/string.indexof8/VB/ixof8.vb#1)]
 
  ]]></format>
```
[the remark this answers is not in the document]
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

File: xml/System/String.xml (hunk 1 above)
Comment (WilliamAntonRohm) [reply xml/System/String.xml]: Done.

**Comment by mairaw** [comment]
New build now that samples change was merged.

**Pull request merged**